### PR TITLE
Feature | Progress bar

### PR DESCRIPTION
**Pull request by davidebukali**
#665

### CODE REVIEW

File: frontend/cypress/fixtures/exportData.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    },
+    "meta": {
+        "delimeter": ",",
+        "fields": ["country_name", "di_id", "iso3", "fragile_2018"],
+        "aborted": false,
+        "linebreak": "\r\n",
+        "truncated": false
+    }
+  }
+
```
Comment (edwinmp) [inline frontend/cypress/fixtures/exportData.json]: End in blank line

File: frontend/src/components/OperationsTableCard/OperationsTableCard.tsx [hunk 2]
```diff
@@ -23,6 +24,9 @@ import { OperationsTableRow } from '../OperationsTableRow';
 import OperationsTableRowActions from '../OperationsTableRowActions';
 import { PaginationRow } from '../PaginationRow';
 import { SearchInput } from '../SearchInput';
+import { toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import { ExportToCsv } from 'export-to-csv';
```
Comment (edwinmp) [inline frontend/src/components/OperationsTableCard/OperationsTableCard.tsx]: This library hasn't been updated in several years so it won't do ... let's keep things in papa parse https://stackoverflow.com/questions/52240221/download-save-csv-file-papaparse

**Review by edwinmp — CHANGES_REQUESTED** [review]
There's another `Export to CSV` button where this feature will need to be applied ... if possible, see if you can build this in a reusable way.

File: frontend/src/components/OperationsTableCard/OperationsTableCard.tsx [hunk 3]
```diff
@@ -159,6 +163,55 @@ const OperationsTableCard: FunctionComponent<OperationsTableCardProps> = (props)
...
+            fieldSeparator: ',',
+            quoteStrings: '"',
+            decimalSeparator: '.',
+            showLabels: true,
+            useTextFile: false,
+            useBom: true,
+            headers: columns && columns.length ? columns : [],
+          });
+
+          toast.update(toastId, { render: 'Success', type: 'success', isLoading: false });
```
Comment (edwinmp) [inline frontend/src/components/OperationsTableCard/OperationsTableCard.tsx]: If possible, account for multiple file being downloaded, so it'd be nice if the success message indicated what was successfully downloaded.

**Comment by edwinmp** [comment]
@davidebukali  I deployed this to staying to see how it works & as expected from looking at the code, it doesn't show progress ... it works pretty much the same way as the old one. Only difference is that it lets you know when the download is complete by popping up an alert. This is not the expected behaviour for this feature.
As with all progress bars (progress indicator in this case), the alert is supposed to show when the process starts and goes away (or gets updated) when the process is complete.

File: frontend/src/components/OperationsTableCard/OperationsTableCard.tsx (hunk 2 above)
Comment (davidebukali) [reply frontend/src/components/OperationsTableCard/OperationsTableCard.tsx]: Removed old library and now using native `js` api

File: frontend/src/components/OperationsTableCard/OperationsTableCard.tsx (hunk 3 above)
Comment (davidebukali) [reply frontend/src/components/OperationsTableCard/OperationsTableCard.tsx]: Updated

File: frontend/cypress/fixtures/exportData.json (hunk 1 above)
Comment (davidebukali) [reply frontend/cypress/fixtures/exportData.json]: Updated

**Review by edwinmp — DISMISSED** [review]
Alert Dean that this is ready for testing on staging ... the e2e tests are failing massively though... I did change the behaviour a bit so that could have broken something, but I think there's more going on overall.

**Comment by edwinmp** [comment]
@davidebukali fix the conflicts on this and let's merge it in ... create a separate ticket for Dean's current bug on large downloads. Comment on the original ticket to that effect & tag the new one. Include the images I sent you in the new ticket.

**Comment by davidebukali** [comment]
@edwinmp when you get the chance, this is ready for another review

**Review by edwinmp — APPROVED** [review]

**Pull request merged**